eth/gasprice/gasprice_test.go

```diff
@@ -111,8 +113,89 @@ func TestSuggestPrice(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to retrieve recommended gas price: %v", err)
 	}
-	expect := big.NewInt(params.GWei * int64(30))
+	expect := big.NewInt(params.GWei)
 	if got.Cmp(expect) != 0 {
 		t.Fatalf("Gas price mismatch, want %d, got %d", expect, got)
 	}
 }
+
+func generateFakeGasPrices(min, max int) []*big.Int {
+	rand.Seed(time.Now().UnixNano())
+	gasPrices := make([]*big.Int, 0)
+	for i := 0; i < 300; i++ {
+		randGasPrice := rand.Intn(max-min+1) + min
+		gasPrices = append(gasPrices, big.NewInt(int64(randGasPrice)))
+	}
+	return gasPrices
+}
+
+func TestRemoveLowOutliers(t *testing.T) {
+	gasPrices := generateFakeGasPrices(180, 220)
+	cpy := make([]*big.Int, len(gasPrices))
+	copy(cpy, gasPrices)
+	// add low gas prices
+	cpy = append(cpy, big.NewInt(5))
+	cpy = append(cpy, big.NewInt(10))
+	cpy = append(cpy, big.NewInt(10))
+	cpy = append(cpy, big.NewInt(10))
+	cpy = append(cpy, big.NewInt(15))
+	cpy = append(cpy, big.NewInt(15))
+	cpy = append(cpy, big.NewInt(15))
+	cpy = append(cpy, big.NewInt(15))
+
+	res := removeOutliers(cpy)
+	// It should remove all the lower gasPrices in the extreme case
+	if len(gasPrices) != len(res) {
```

Comment:
"If the removed result length is not equal to the original gasPrices length, then error."

This doesn't make sense.